django/db/migrations/autodetector.py

```diff
@@ -52,21 +52,33 @@ def deep_deconstruct(self, obj):
         Used for full comparison for rename/alter; sometimes a single-level
         deconstruction will not compare correctly.
         """
-        if not hasattr(obj, 'deconstruct') or isinstance(obj, type):
-            return obj
-        deconstructed = obj.deconstruct()
-        if isinstance(obj, models.Field):
-            # we have a field which also returns a name
-            deconstructed = deconstructed[1:]
-        path, args, kwargs = deconstructed
-        return (
-            path,
-            [self.deep_deconstruct(value) for value in args],
-            {
+        if isinstance(obj, list):
+            items = [self.deep_deconstruct(value) for value in obj]
```

Comment:
Argh! Sorry for the brainfart... better tests added in a1435fa695c83d7207c8b3156d0c8c26ca186a78, fixed in e511c330fca560b7f73d584fd43f55e420783279.